### PR TITLE
gborah-OKTA-413579_Minor_improvements_on_OIE_policy_types_docs

### PR DESCRIPTION
**Pull request by gautamborah-okta**
## Description:
- **What's changed?** 

1. Profile Enrollment policy description updated.
2. App Sign On Policy description updated and HAL link is described.
3. Office365Client Condition removed - Office 365 team decided NOT to document that condition. They don’t want to support this via APIs.
4. Expression Language Condition added
5. Profile Enrollment spelling corrected.

- **Is this PR related to a Monolith release?** 

This is OIE related changes, not related to any release.

### Resolves:

* [OKTA-413579](https://oktainc.atlassian.net/browse/OKTA-413579)

### CODE REVIEW

**Comment by unniravindranathan-okta** [comment]
looks good to me!

File: packages/@okta/vuepress-site/docs/reference/api/policy/index.md [hunk 1]
```diff
@@ -1038,6 +1038,25 @@ Specifies a particular level of risk to match on
 }
 ```
 
+#### Expression Language Condition object
+
+Use Okta Expression Language as a condition.
+For details please see [Okta Expression Language in Identity Engine](/docs/reference/okta-expression-language-in-identity-engine/)
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/policy/index.md:1044]:
```suggestion
See [Okta Expression Language in Identity Engine](/docs/reference/okta-expression-language-in-identity-engine/)
```

File: packages/@okta/vuepress-site/docs/reference/api/policy/index.md [hunk 2]
```diff
@@ -1706,6 +1725,8 @@ You can define multiple IdP instances in a single Policy Action. This allows use
 
 The app sign-on policy determines the extra levels of authentication (if any) that must be performed before you can invoke a specific Okta application. It is always associated with an app through a Mapping. Okta Identity Engine always evaluates both the Okta sign-on policy and the sign-on policy for the app. The resulting user experience is the union of both policies. App sign-on policies have the type `ACCESS_POLICY`.
 
+When you create a new application, it's created with a new app sign on policy by default. You can't create a new app sign on policy and assign it to the application. Instead, consider editing the default one to meet your needs.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/policy/index.md:1728]:
```suggestion
When you create a new application, it's created with a new app sign-on policy by default. You can't create a new app sign-on policy and assign it to the application. Instead, consider editing the default one to meet your needs.
```

File: packages/@okta/vuepress-site/docs/reference/api/policy/index.md [hunk 3]
```diff
@@ -1938,7 +1969,9 @@ The number of Authenticator class constraints in each Constraint object must be
 
 > **Note:** This feature is only available as a part of Okta Identity Engine. [Contact support](mailto:dev-inquiries@okta.com) for further information.
 
-Profile Enrollemnt policies specify which profile attributes are required for creating new Users through self-service registration, and also can be used for progressive profiling. The type is specified as `PROFILE_ENROLLMENT`.
+Profile Enrollment policies specify which profile attributes are required for creating new Users through self-service registration, and also can be used for progressive profiling. The type is specified as `PROFILE_ENROLLMENT`.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/policy/index.md:1972]:
```suggestion
Profile Enrollment policies specify which profile attributes are required for creating new Users through self-service registration and also can be used for progressive profiling. The type is specified as `PROFILE_ENROLLMENT`.
```

**Review by susanharper-okta — COMMENTED** [review]
couple nits that I'll commit myself

**Review by susanharper-okta — APPROVED** [review]
LGTM

**Pull request merged**